src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -125,20 +124,55 @@ private void AddRuleset()
             }
         }
 
-        private void AddAdditionalFile()
+        private void EnsureAdditionalFileIsReferenced()
         {
-            try
+            var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+            projectSystem.AssertLocalServiceIsNotNull();
+
+            // We assume that we've checked during the "Prepare" step that there are references to other
```

Comment:
There's enough complexity and weirdness around what does and doesn't worth that I though it worth the extensive documentation.